float64 data
string from_node
float64 timestamp